pkg/aws/policies.go

```diff
@@ -729,18 +729,40 @@ func checkIfROSAOperatorRole(roleName *string, credRequest map[string]*cmv1.STSO
 
 func (c *awsClient) DeleteOperatorRole(roleName string) error {
 	role := aws.String(roleName)
-	err := c.detachOperatorRolePolicies(role)
+	policies, err := c.getAttachedOperatorRolePolicies(role)
+	if err != nil {
+		return err
+	}
+	err = c.detachOperatorRolePolicies(role)
 	if err != nil {
 		if aerr, ok := err.(awserr.Error); ok {
 			switch aerr.Code() {
 			case iam.ErrCodeNoSuchEntityException:
 				fmt.Printf("Policies does not exists for role '%s'",
 					roleName)
+			case iam.ErrCodeDeleteConflictException:
+				err2 := c.DeleteRole(roleName, role)
```

Comment:
I'm not sure I like this pattern. It duplicates code. I think a better idea might be to explicitly ignore this error:
```go
err = c.detachOperatorRolePolicies(role)
if err != nil {
  if aerr, ok := err.(awserr.Error); ok {
    switch aerr.Code() {
    case iam.ErrCodeNoSuchEntityException:
      fmt.Printf(...)
      err = nil
    case iam.ErrCodeDeleteConflictException:
      fmt.Printf(...)
      err = nil
    }
  }
  if err != nil {
    return err
  }
}
err = c.DeleteRole(roleName, role)
```
The idea being that these two error codes should not prevent ROSA from attempting to delete the role. The role deletion might still succeed for all we know, and these policy errors should either fail the entire thing, or just be logged and ignored.

Reply:
That's a much better way of handling this, implemented now